.evergreen/legacy_config_generator/evergreen_config_generator/functions.py

```diff
@@ -17,96 +17,106 @@
 
 from evergreen_config_generator import ConfigObject
 
+from . import Value, MutableValueMapping, ValueMapping, ValueOrderedDict
 
-def func(func_name, **kwargs):
-    od = OD([('func', func_name)])
+
+def func(func_name: str, **kwargs: Value) -> MutableValueMapping:
+    od: MutableValueMapping = OD([("func", func_name)])
     if kwargs:
-        od['vars'] = OD(sorted(kwargs.items()))
+        od["vars"] = OD(sorted(kwargs.items()))
 
     return od
 
 
-def s3_put(remote_file, project_path=True, **kwargs):
+def s3_put(remote_file: str, project_path: bool = True, **kwargs: Value) -> ValueMapping:
     if project_path:
-        remote_file = '${project}/' + remote_file
-
-    od = OD([
-        ('command', 's3.put'),
-        ('params', OD([
-            ('aws_key', '${aws_key}'),
-            ('aws_secret', '${aws_secret}'),
-            ('remote_file', remote_file),
-            ('bucket', 'mciuploads'),
-            ('permissions', 'public-read')]))])
-
-    od['params'].update(kwargs)
-    return od
-
-
-def strip_lines(s):
-    return '\n'.join(line for line in s.split('\n') if line.strip())
-
-
-def shell_exec(script,
-               test=True,
-               errexit=True,
-               xtrace=False,
-               silent=False,
-               continue_on_err=False,
-               working_dir=None,
-               background=False,
-               add_expansions_to_env=False,
-               redirect_standard_error_to_output=False,
-               ):
-    dedented = ''
+        remote_file = "${project}/" + remote_file
+
+    return ValueOrderedDict(
+        [
+            ("command", "s3.put"),
+            (
+                "params",
+                ValueOrderedDict(
+                    (
+                        ("aws_key", "${aws_key}"),
+                        ("aws_secret", "${aws_secret}"),
+                        ("remote_file", remote_file),
+                        ("bucket", "mciuploads"),
+                        ("permissions", "public-read"),
+                        *kwargs.items(),
+                    )
+                ),
+            ),
+        ]
+    )
```

Comment:
The explicit type is required to hint the checker about the valid value types for the object, as otherwise it generates a `dict[str, str]`, which cannot be inserted into `cmd` because of type variance conflicts.